### PR TITLE
[setup] add basic macos tools

### LINKED ISSUE #2331: script/bootstrap or instructions to install tools should be available for Mac
The openthread.io site mentions running `script/bootstrap` in order to automatically install required tools, but it doesn't work on Mac:

```
$ ./script/bootstrap 
./script/bootstrap: line 95: .: filename argument required
.: usage: . filename [arguments]
```

### PR DESCRIPTION
**Pull request by jwhui**
Resolves #2331.

### CODE REVIEW

**Comment by turon** [comment]
Does it make sense to add `sudo easy_install pip` to the script, so the machine ready to install packages needed by the thread-cert test suite?

**Comment by itavero** [comment]
I've just tried this. The tooling looks alright, however I do get an error when trying to compile the latest `master` branch. I'm not sure if it is related.

```
/path/to/openthread/examples/../src/core/common/instance.cpp: In static member function 'static ot::Instance& ot::Instance::Get()':
/path/to/openthread/examples/../src/core/common/instance.cpp:96:55: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
     return *reinterpret_cast<Instance *>(&sInstanceRaw);
```

I get this with both the CC2538 and the CC2650 examples. I have not yet tried any other examples at this point in time.

**Comment by jwhui** [comment]
@itavero, what version of `arm-none-eabi-gcc` are you using?

File: script/bootstrap [hunk 1]
```diff
@@ -67,7 +67,17 @@ install_packages_rpm()
 
 install_packages_brew()
 {
-    echo 'macOS not supported currently' && false
+    # add autotools
+    brew install automake libtool
+
+    # add ARM toolchain
+    brew tap PX4/homebrew-px4
+    brew install gcc-arm-none-eabi
```
Comment (beriberikix) [inline script/bootstrap]: I believe this install [v5.4](https://github.com/PX4/homebrew-px4/blob/master/gcc-arm-none-eabi.rb#L7) but that may change overtime. Should this install a specific version?

Comment (jwhui) [reply script/bootstrap]: We intentionally do not prescribe a particular release.  On [Travis](https://travis-ci.org/openthread/openthread), we test against the 4, 5, and 6 series.

Comment (beriberikix) [reply script/bootstrap]: Fair, only thing is it may result in more issues :)

File: script/bootstrap [hunk 2]
```diff
@@ -67,7 +67,17 @@ install_packages_rpm()
...
+
+    # check for gcc for posix examples
+    if ! which gcc; then
+        echo 'warning: clang/gcc needed for posix examples - please install xcode to use posix examples'
```
Comment (beriberikix) [inline script/bootstrap]: They actually just need the CLI tool. You can link to apple: https://developer.apple.com/download/more/

**Comment by jwhui** [comment]
@itavero, submitted a proposed fix in #2361.

**Comment by itavero** [comment]
@jwhui I'm using v6.3.1 apparently. I'll pull in the changes from master and check if it works now. Thanks!

```
arm-none-eabi-gcc (GNU Tools for ARM Embedded Processors 6-2017-q2-update) 6.3.1 20170620 (release) [ARM/embedded-6-branch revision 249437]
```

**Comment by itavero** [comment]
I tried to compile the same examples as before. This time they all finished without any errors. 👍

**Pull request merged**